index.js

```diff
@@ -124,14 +134,16 @@ types[MAESTRO] = {
 };
 
 function creditCardType(cardNumber) {
-  var type, value;
+  var type, value, i;
   var result = [];
 
   if (!(typeof cardNumber === 'string' || cardNumber instanceof String)) {
     return result;
   }
 
-  for (type in types) {
+  for (i = 0; i < testOrder.length; i++) {
+    type = testOrder[i];
+
     if (!types.hasOwnProperty(type)) { continue; }
```

Comment:
This line isn't necessary any more, right? `types` should always have the `type` property?